From 2fa7dc9efcf4198d4f78268d22f68747045dafec Mon Sep 17 00:00:00 2001
From: wfeng6 <wei.feng@intel.com>
Date: Wed, 21 Nov 2012 06:12:36 +0800
Subject: [PORT FROM MAIN] Enable VAVideoDecoder as thumbnail generation path

BZ: 48580

Use Intel HW decoder as the thumbnail generation path to improve
performance.

DO NOT UPSTREAM!

Change-Id: I9c32367e972a43dd1cc50ef6d3d5adde81aa3fe1
Orig-Change-Id: Ia5b4d533fe30985e234128e871e29e0bf4f04970
Signed-off-by: wfeng6 <wei.feng@intel.com>
Reviewed-on: http://android.intel.com:8080/74898
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Tong, BoX <box.tong@intel.com>
Reviewed-by: Tong, BoX <box.tong@intel.com>
---
 media/libstagefright/Android.mk   |   11 +++++++++++
 media/libstagefright/OMXCodec.cpp |   11 +++++++++++
 2 files changed, 22 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/Android.mk b/media/libstagefright/Android.mk
index 560052a..fdeec2f 100644
--- a/media/libstagefright/Android.mk
+++ b/media/libstagefright/Android.mk
@@ -114,6 +114,17 @@ LOCAL_STATIC_LIBRARIES += \
 LOCAL_CPPFLAGS += -DUSE_INTEL_MDP
 endif
 
+ifeq ($(USE_INTEL_VA),true)
+LOCAL_C_INCLUDES += $(TARGET_OUT_HEADERS)/libvavideodecoder
+LOCAL_STATIC_LIBRARIES += libvavideodecoder
+LOCAL_SHARED_LIBRARIES += libva \
+                          libva-android \
+                          libva-tpi \
+                          libva_videodecoder \
+                          libmixvbp
+LOCAL_CPPFLAGS += -DUSE_INTEL_VA
+endif
+
 LOCAL_SRC_FILES += \
         chromium_http_stub.cpp
 LOCAL_CPPFLAGS += -DCHROMIUM_AVAILABLE=1
diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index 1b975ef..abcffc6 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -53,6 +53,10 @@
 #include "MetaDataExt.h"
 #endif
 
+#ifdef USE_INTEL_VA
+#include "VAVideoDecoder.h"
+#endif
+
 namespace android {
 
 // Treat time out as an error if we have not received any output
@@ -339,6 +343,13 @@ sp<MediaSource> OMXCodec::Create(
         }
 #endif
 
+#ifdef USE_INTEL_VA
+        if (!strncmp(componentName, "VAVideoDecoder", 14)) {
+            ALOGV("Loading VAVideoDecoder");
+            return new VAVideoDecoder(source);
+        }
+#endif
+
         if (createEncoder) {
             sp<MediaSource> softwareCodec =
                 InstantiateSoftwareEncoder(componentName, source, meta);
-- 
1.7.4.1

